src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -99,36 +125,81 @@ private void ReplayErrors(AggregateException exceptions)
 
         private void CheckPackageVulnerabilities(IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities)
         {
-            Dictionary<PackageIdentity, Dictionary<PackageVulnerabilityInfo, List<string>>>? packagesWithKnownVulnerabilities =
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            Dictionary<PackageIdentity, PackageAuditInfo>? packagesWithKnownVulnerabilities =
                 FindPackagesWithKnownVulnerabilities(knownVulnerabilities);
+            stopwatch.Stop();
+            CheckPackagesDurationSeconds = stopwatch.Elapsed.TotalSeconds;
 
             if (packagesWithKnownVulnerabilities == null) return;
 
+            stopwatch.Restart();
+
+            int directPackageCount = packagesWithKnownVulnerabilities.Values.Count(p => p.IsDirect);
```

Comment:
It seems we're using the Count function to determine the size of the lists we've created below. Is there an alternative approach to calculate the number of elements?

Reply:
Nothing better seems obvious to me.